### DISCUSSION TITLE
How about some love for docker swarm

### DISCUSSION THREAD

**Discussion by dazinator** [docker/extension-ideas#151] [Ideas]
Sure most large enterprises use k8's now, but swarm is simple and good for simpler use cases - unless anyone knows different?

There is a severe lack of tooling or projects for swarm especially around debugging or workflows.

- Seeing lots of "bridges" for kubernetes. These basically deploy proxy service into the cluster so that traffic is forwarded to your local machine where you can debug locally as if your service was running in the environment receiving traffic. This capabilty would also be super useful when targetting docker swarm but I don't think anything exists.

- Secrets and configs can be held in many places these days - I am thinking of Azure KeyVault and other cloud vender equivalents. How about a service that synchronises docker config/s & secrets from one or many of these locations? This way you can sync them from your cloud provider's service, but then your containers can directly use  docker configs and secrets without having to worry about intergation with azure keyavault or anything else. 

I am actually working on a side project that does the second thing - the config / secret sync, initially one way (download) from KeyVault.

**Comment by s4ke** [comment]
I think something that can serve as a starter for this discussion would be a Swarm extension that allows to spin up a Docker Swarm inside of Docker containers in a single click locally for testing. Something like https://gist.github.com/pecigonzalo/66905b0839b88ce894eaf44846b85d65 but with a nicer UI and more insights.

We would have to check the interactions with the Portainer extension, maybe we don't have to build out everything for local cluster management ourselves.
